tasks/libs/common/docker_hub.py

```diff
@@ -0,0 +1,16 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry  # type: ignore
+
+
+def get_latest_version(repository, namespace="datadog") -> str:
+    session = requests.session()
+    retries = Retry(total=3, backoff_factor=1)
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+
+    response = session.get(f"https://hub.docker.com/v2/namespaces/{namespace}/repositories/{repository}/tags")
+    response.raise_for_status()
+
+    return sorted(
+        [result["name"] for result in response.json()["results"] if result["name"] != "latest"], reverse=True
+    )[0]
```

Comment:
`sorted` will sort the versions by lexicographical order.
It means that `7.100.0` will be considered as being before `7.55.2` because `1` is before `5` whereas it should be considered as being after because `100` is greater than `55`.

We should most probably leverage [`semver`](https://pypi.org/project/semver/) to compare the versions.

Also, fully sorting a list to get only the max value is quite inefficient.

Here is an example of code to find the highest version in a list: https://github.com/DataDog/datadog-agent/blob/831ef1d8a4c1b31007c020088ab9856835c78650/tasks/agent.py#L465-L472

Reply:
It was not that hard indeed https://github.com/DataDog/datadog-agent/pull/28146/commits/4a40123311a55e063dde12e4dceb56ee56dca65b, thanks!